src/adapters/server-adapter.ts

```diff
@@ -99,7 +106,10 @@ export async function handler(
   );
   response.headers = { ...response.headers, ...middlewareResponseHeaders };
 
-  debug("response headers", response.headers);
+  // new beta appDir does not set statusCode and lambda fails to correctly send response
```

Comment:
This appears to be a bug w/ app dir routes. The root route has the statusCode set but all nested routes are undefined.

Reply:
might be fixed: https://github.com/vercel/next.js/pull/47096